.github/workflows/prod-deploy.yml

```diff
@@ -0,0 +1,33 @@
+name: Production Deployment Workflow
+on:
+  pull_request:
+    branches: [master]
+      types: [closed]
+jobs:
+  deploy_job:
+    runs-on: ubuntu-latest
+    steps:
+    - name: deploy to production server with ssh
+      uses: appleboy/ssh-action@master
+      with:
+        host: ${{secrets.PROD_HOST}}
+        username: ${{secrets.PROD_USER}}
+        key: ${{secrets.PROD_SSH_KEY}}
+        port: ${{secrets.PROD_PORT}}
+        script: |
+          sudo -iu cosmos
+          cd /opt/cosmos/Cosmos
+          git pull
+          pipenv sync
```

Comment:
I believe you need to use a certain flag to force it to only install, not upgrade. But I might be wrong about that.

Reply:
that's what [pipenv sync does ](https://pipenv.pypa.io/en/latest/advanced/#using-pipenv-for-deployments)